game-core/src/main/java/games/strategy/triplea/attachments/CanalAttachment.java

```diff
@@ -163,4 +168,32 @@ public void validate(final GameData data) throws GameParseException {
       throw new GameParseException("Canal named " + m_canalName + " must have landTerritories set!" + thisErrorMsg());
     }
   }
+
+  private static Map<String, Function<IAttachment, AttachmentProperty<?>>> getPopulatedAttachmentMap() {
+    return ImmutableMap.<String, Function<IAttachment, AttachmentProperty<?>>>builder()
+        .put("canalName", ofCast(a -> AttachmentProperty.of(a::setCanalName, a::getCanalName, a::resetCanalName)))
+        .put("landTerritories",
+            ofCast(a -> AttachmentProperty.of(
+                a::setLandTerritories,
+                a::setLandTerritories,
+                a::getLandTerritories,
+                a::resetLandTerritories)))
+        .put("excludedUnits",
+            ofCast(a -> AttachmentProperty.of(
+                a::setExcludedUnits,
+                a::setExcludedUnits,
+                a::getExcludedUnits,
+                a::resetExcludedUnits)))
+        .build();
+  }
+
+  @Override
+  public Map<String, Function<IAttachment, AttachmentProperty<?>>> getAttachmentMap() {
+    return attachmentSetters;
```

Comment:
Is this intentional to call it `attachmentSetters`? Seems out of place to not be called `attachmentMap`

Reply:
Not really, I did a bad naming choice when starting and just copied those 2 lines everywhere 😅